src/components/SearchBox.vue

```diff
@@ -1,75 +1,235 @@
+        timeTo:'',
+        instanceName: "",
+        timeFrom:'',
+        searchOptions:{}
+    }),
     computed: {
       ...mapState({
         queryDisplay: state => state.searchStore.queryDisplay,
         query: state => state.searchStore.query,
         loading: state => state.searchStore.loading,
         instance: state => state.searchStore.instance
-
       })
     },
     created() {
-      this.searchQuery = this.queryDisplay;
+      //A fast check on weather we do have a query set - and if we dont, we check if the params hae a query, and we get that.
+      //If all fails, fallback is "No query."
+      if(this.searchQuery == "") {
+        this.queryDisplay !== "" ? this.searchQuery = this.queryDisplay :
+        this.$router.history.current.params.query ? 
+        this.searchQuery = this.$router.history.current.params.query.substr(0, this.$router.history.current.params.query.indexOf('&')) ||
+        this.$router.history.current.params.query 
+        : "No query."
+      }
+      //We need the search options, so we check if the key is present and identifiable in either the instance or the route
+      //Then we get the search options from the meloarInstances opbject.
+        this.MeloarInstances.instances.filter(item => {
+        item.key === this.instance || item.key === this.$route.params.instance ? (this.instanceName = item.name,
+                                                                                  this.searchOptions = item.searchOptions) : null
+      });
+      //When creating, check if we already have the time for the slider in the params, and if so, put them into the slider
+      if(this.time === true) {
+        if(this.$router.history.current.params.query) {
+        this.$router.history.current.params.query.split("&fq=").filter(item => {
+          item.includes("ff_primaryobject_year_from_i") ? this.timeFrom = parseInt(this.$_returnTimeFromQueryString(item)) : null
+          item.includes("ff_primaryobject_year_to_i") ? this.timeTo = parseInt(this.$_returnTimeFromQueryString(item)) : null
+          })
+        }
+        //Double check - if we didn't get anything from the params, we set them ourselves from the options.
+        this.timeFrom === '' ? this.timeFrom = this.searchOptions.timeOptions.min : null;
+        this.timeTo === '' ? this.timeTo = this.searchOptions.timeOptions.max : null;
+      }
     },
     watch: {
       queryDisplay: function(newValue) {
       this.searchQuery = newValue
+    },
+    $route(to) {
+      //Make sure, when the route is updated in any way - that we update the displayQuery and query.
+      //This makes sure, that pressing the backbutton in the browser rightly updates the used query in the searchbox.
+      console.log(to.params)
+      this.updateQuery(to.params.query);
+      to.params.query.indexOf("&d=") !== -1 && to.params.query.indexOf("&d=") < to.params.query.indexOf("&fq=") ? this.updateQueryDisplay(to.params.query.split("&d=")[0]) :
```

Comment:
This is actually some really defensive programming, that's not strictly necessary right now, because you can't filter on map after a search, so if a location param is set, it'll always be first.

So the check first check is not really necessary right now. If we ever allow a filtering based on map, something like this will be. I can rewrite it in a better way tho, so the check makes more sense.